### PR TITLE
Feature: Add Possibility to Register Uri as Bitmap

### PR DESCRIPTION
**Pull request by inforithmics**
Part of this
https://github.com/Mapsui/Mapsui/pull/2581

Things done:
1. Adds to the BitmapRegistry the possibility to Add Bitmaps as Uri with the format of this https://github.com/Mapsui/Mapsui/issues/1998 issue.
2. Made BitmapRegistry Threadsafe
3. Added more Unit Tests

The next pull request will be implementing this.
 https://github.com/Mapsui/Mapsui/issues/1998

And the https://github.com/Mapsui/Mapsui/pull/2581 pull request should then only be changing the samples to use the Icon/BitmapPath.

### CODE REVIEW

**Comment by pauldendulk** [comment]
Without having to look at the code I would like to understand what it means that the registry is hierarchical . And what is the reason to add this functionality. Do do you have a sample where this would be helpful?

**Comment by inforithmics** [comment]
> Without having to look at the code I would like to understand what it means that the registry is hierarchical . And what is the reason to add this functionality. Do do you have a sample where this would be helpful?

The Hierarchical part is needed. So that there is a static BitmapRegistry.Instance for registering Bitmaps for maps and a dynamic BitmapRegistry Instance in the RenderService. The one in the RenderService is disposed the static one not.

**Comment by inforithmics** [comment]
An alternative implementation could be a RenderServiceBitmapRegistry that is only used in the RenderService.

**Comment by inforithmics** [comment]
@pauldendulk for easier review I removed the Hierarchical part of the BitmapRegistry. Bitmap Registry is now threadsafe and if a get with an invalid bitmapid is done an exception is thrown.

File: Mapsui/Styles/BitmapRegistry.cs [hunk 1]
```diff
@@ -39,13 +48,77 @@ public int Register(object bitmapData, string? key = null)
...
+        switch (bitmapPath.Scheme)
+        {
+            case "embeddedresource":
+                if (resourceCache.TryGetValue(bitmapPath.Host, out var found))
+                {
+                    stream = found.assembly.GetManifestResourceStream(found.realResourceName);
+                }
+                else
+                {
+                    foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
```
Comment (pauldendulk) [inline Mapsui/Styles/BitmapRegistry.cs:64]: I realized just the problem that if you want to load an embedded resource you also need to know the specific assembly. I had just started thinking about the problem, but now I see you already solved this. Great.

**Review by pauldendulk — APPROVED** [review]
Great, thanks a lot.

**Pull request merged**